lgc/include/lgc/state/ResourceUsage.h

```diff
@@ -280,12 +280,9 @@ struct ResourceUsage {
 
   // Usage of generic input/output
   struct {
-    // Map from shader specified locations to tightly packed locations
-    std::map<unsigned, unsigned> inputLocMap;
-    std::map<unsigned, unsigned> outputLocMap;
-
-    // The original and new InOutLocations for shader cache
-    std::map<unsigned, unsigned> inOutLocMap;
+    // Map from shader specified InOutLocationInfos to tightly packed InOutLocationInfos
+    std::map<unsigned, unsigned> inputLocInfoMap;
+    std::map<unsigned, unsigned> outputLocInfoMap;
```

Comment:
As an additional improvement, can we avoid the use of the effectively untyped `unsigned` here? At least the value type of this map could probably be changed to `InOutLocationInfo` quite easily, and would avoid the need to explicitly refer to the u16All member.

Can we also change the key type to `InOutLocationInfo`? This would require an operator< in the class so that it works with std::map.

Reply:
Okay, I will change the key type and value type of the two maps.

Reply:
Done. Is it proper to use InvalidValue (unsigned) to initialize the value of the map?